### PR TITLE
Spark Configuration can be dynamically updated or overriden

### PR DESCRIPTION
**Pull request by kumare3**
# TL;DR
A user can now easily override spark and hadoop conf using dynamic tasks or when registering a workflow. This does not solve overrides completely, but is step 1 towards it

```python
@spark_task(spark_conf={...}, hadoop_conf={...})
def foo(...):
  pass

@dynamic_task
def dynamic():
  t = foo.with_overrides(spark_conf={})()
  yield t
```
## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
SdkSparkTask has a special method called with_conf(...) that creates a new flyte task with the override configuration.

## Tracking Issue
https://github.com/lyft/flyte/issues/528

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/common/tasks/sdk_runnable.py [hunk 1]
```diff
@@ -222,6 +222,7 @@ def __init__(
         timeout,
         environment,
         custom,
+        copy=False,
```
Comment (wild-endeavor) [inline flytekit/common/tasks/sdk_runnable.py:225]: did you want to set this?

Comment (kumare3) [reply flytekit/common/tasks/sdk_runnable.py:225]: ohh no i should remove that, oops

File: flytekit/common/tasks/spark_task.py [hunk 2]
```diff
@@ -165,3 +169,36 @@ def _get_container_definition(self, **kwargs):
...
+        Creates a new SparkJob instance with the modified configuration or timeouts
+        """
+        if not new_spark_conf:
+            new_spark_conf = self._spark_job.spark_conf
+
+        if not new_hadoop_conf:
+            new_hadoop_conf = self._spark_job.hadoop_conf
+
+        updates = {"spark": new_spark_conf, "hadoop": new_hadoop_conf}
+        salt = _hashlib.md5(_json.dumps(updates, sort_keys=True).encode("utf-8")).hexdigest()
```
Comment (EngHabu) [inline flytekit/common/tasks/spark_task.py:185]: use the entire job?

File: flytekit/common/tasks/spark_task.py [hunk 3]
```diff
@@ -165,3 +169,36 @@ def _get_container_definition(self, **kwargs):
     def _get_kwarg_inputs(self):
         # Trim off first two parameters as they are reserved for workflow_parameters and spark_context
         return set(_getargspec(self.task_function).args[2:])
+
+    def with_conf(self, new_spark_conf: typing.Dict[str, str] = None,
```
Comment (EngHabu) [inline flytekit/common/tasks/spark_task.py:173]:
```suggestion
    def with_overrides(self, new_spark_conf: typing.Dict[str, str] = None,
```

**Comment by wild-endeavor** [comment]
also make lint

File: flytekit/__init__.py [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 import flytekit.plugins  # noqa: F401
 
-__version__ = "0.13.0b7"
+__version__ = "0.13.0b8"
```
Comment (wild-endeavor) [inline flytekit/__init__.py:5]:
```suggestion
__version__ = "0.13.0b9"
```

Comment (wild-endeavor) [reply flytekit/__init__.py:5]: i'm gonna use 8

**Review by wild-endeavor — APPROVED** [review]

**Review by wild-endeavor — APPROVED** [review]

**Review by wild-endeavor — APPROVED** [review]
